### ISSUE TITLE
Améliorer le nommage et les chemins des paramètres

### ISSUE THREAD

**Issue by sandcha** [openfisca/openfisca-france#822]

**Comment by fpagnoux** [comment]
Autre problème : je tappe BMAF dans le moteur de recherche du legislation explorer, je suis submergé par des résultats qui contiennent `en_de_la_bmaf​`. Ce genre d'info devrait être dans le description, pas dans le titre.

**Comment by benjello** [comment]
@sandcha @fpagnoux @michelbl : voulez-vous un mapping noms longs -> noms courts ou des règles plus élaborées ?

cc @bfabre01 @ClaireLeroyIPP

**Comment by michelbl** [comment]
Oui je pense qu'il faudra utiliser un mapping pour racourcir automatiquement lors de l'import des baremes IPP.

**Comment by benjello** [comment]
Je vais faire cela mais il me semble que ce que cette issue montre aussi que créée un nouveau répertoire par niveau de profondeur de l'arbre peut être inutilement systématique.
On peut s'arrêter à 3 niveaux (ex. `parameters/taxation_capital/pfl_av`) et garder le reste dans un même fichier yaml. Qu'en pensez-vous ?

**Comment by benjello** [comment]
Il y a un autre problème potentiel. Si je raccourci certains noeuds intermédiaires, il faut le faire pour le noeud donc tous les éléments du noeud. Bref cela plaide pour la solution évoquée plus haut.
Dans un deuxième temps, cela vaudra le coup de nettoyer les noms plus systématiquement (c'est vraiment assez chiant pour ne pas avoir à faire deux fois la même chose de façon incohérente).

Qu'en pensez-vous @michelbl @sandcha @fpagnoux ?

**Comment by michelbl** [comment]
> On peut s'arrêter à 3 niveaux (ex. parameters/taxation_capital/pfl_av) et garder le reste dans un même fichier yaml. Qu'en pensez-vous ?

Pas d'avis là dessus

> Il y a un autre problème potentiel. Si je raccourci certains noeuds intermédiaires, il faut le faire pour le noeud donc tous les éléments du noeud. Bref cela plaide pour la solution évoquée plus haut.
> Dans un deuxième temps, cela vaudra le coup de nettoyer les noms plus systématiquement (c'est vraiment assez chiant pour ne pas avoir à faire deux fois la même chose de façon incohérente).

Je n'ai pas bien compris. Pourrais-tu donner un exemple stp ?

**Comment by benjello** [comment]
Si je raccourci le noeud intermédiaire `bons_ou_contrats_de_capitalisation_et_placements_de_meme_nature_assurance_vie_lors_du_denouement_du_contrat` qui me dit que ce n'est pas aussi présent dans un autre chemin qui passe sous la barre des 150 et du coup je crée une divergence qui n'a pas lieu d'être.

**Comment by michelbl** [comment]
Oui je suis d'accord avec toi. Il faut vérifier, après tout renommage, que l'ancien nom n'est plus présent nulle part.

**Comment by benjello** [comment]
@sandcha : a-t-on résolu ce problème ?

**Comment by fpagnoux** [comment]
J'ai l'impression que les noms très longs sont encore présents dans OpenFisca, donc a priori non. 

https://github.com/openfisca/openfisca-france/issues/819, qui empêchait de cloner sous Windows, a par contre été résolu.

**Comment by sandcha** [comment]
Oui, ça n'a pas évolué. On a réduit les chances d'être bloqués par cette problématique sous windows (en réduisant les paths de paramerters/.../...yaml à 150 caractères max) mais on ne l'a pas éliminée.

@benjello, sur cette question :
> On peut s'arrêter à 3 niveaux (ex. parameters/taxation_capital/pfl_av) et garder le reste dans un même fichier yaml. Qu'en pensez-vous ?

Si les noms de répertoire du type `imposition_des_plus_values_mobilieres_levees_d_options_attribuees_depuis_27_4_2000_cessions_realisees_moins_de_deux_ans_apres_la_levee` proviennent d'une chaîne automatique d'import de valeurs, le fait de s'arrêter à 3 niveaux me semble effectivement être le choix avec le moins d'effet en amont.

Ceci dit, je suis d'accord avec @fpagnoux sur :
> Autre problème : je tappe BMAF dans le moteur de recherche du legislation explorer, je suis submergé par des résultats qui contiennent en_de_la_bmaf​. Ce genre d'info devrait être dans le description, pas dans le titre.

Or, réduire à 3 niveaux, n'améliorerait pas la lisibilité en aval/côté legislation explorer. 

Donc, avant d'envisager une éventuelle réduction automatique de tous les chemins, je serais pour un mapping de renommage des répertoires aux noms les plus longs (calcul encore à faire pour identifier la longueur adaptée). Ce mapping serait ensuite repris par un script.
